File: src/main/java/com/mgjg/kmztracker/KmzTrackerApp.java
package com.mgjg.kmztracker;

import android.app.Application;

public class KmzTrackerApp extends Application
{

    private static KmzTrackerApp instance;

    @Override
    public void onCreate()
    {
        super.onCreate();
        instance = this;
        AppPreferences.makeInstance(this);
    }

    KmzTrackerApp getInstance()
    {
        return instance;
    }
}

File: src/main/java/com/mgjg/kmztracker/KmzTrackerApp.kt
package com.mgjg.kmztracker

import android.app.Application

class KmzTrackerApp : Application() {

  override fun onCreate() {
    super.onCreate()
    instance = this
    AppPreferences.makeInstance(this)
  }

  fun getInstance(): KmzTrackerApp? {
    return instance
  }

  companion object {

    private var instance: KmzTrackerApp? = null
  }

}
